### ISSUE TITLE
Nested field names containing underscore can no longer be parsed from env

### ISSUE THREAD

**Issue by WieslerAA** [pydantic/pydantic-settings#451]
Example settings that worked before `2.6.0`:

```py
from pydantic_settings import BaseSettings, SettingsConfigDict

os.setenv("INNER_TEST_STR", "abc")

class Inner(BaseSettings):
    model_config = SettingsConfigDict(env_prefix="INNER_")
    test_str: str

class Settings(BaseSettings):
    inner: Inner
```

This can't be done any more. The env_prefix of `Inner` is completely ignored, since there is no env variable for `inner`.

Note that I also can't use the env separator `_` since then the env variable is parsed as `inner.test.str` which is wrong.

**Comment by hramezani** [comment]
Thanks for reporting this. 

It is a duplicate of https://github.com/pydantic/pydantic-settings/issues/447


I close this issue

**Comment by hramezani** [comment]
I [reverted the fix that caused the problem](https://github.com/pydantic/pydantic-settings/commit/e73b349e3b9b0d40b599b1314d6759abaa2ab87c) and released  [pydantic-settings 2.6.1](https://pypi.org/project/pydantic-settings/).
